src/mono/wasm/runtime/library_mono.js

```diff
@@ -1224,6 +1233,32 @@ var MonoSupportLib = {
 			return getter_res.length > 0 ? getter_res [0] : {};
 		},
 
+		/**
+		 * @param  {string} objectIdStr objectId
+		 * @param  {string} name property name
+		 * @returns {object} return true if it works and false if it doesn't
+		 */
+		_set_value_on_object: function (objectIdStr, name, newvalue) {
+			const id = this._parse_object_id (objectIdStr);
+			if (id === undefined)
+				throw new Error (`Invalid object id: ${objectIdStr}`);
+
+			let getter_res;
+			if (id.scheme == 'object') {
+				if (isNaN (id.o) || id.o < 0)
+					throw new Error (`Invalid object id: ${objectIdStr}`);
+
+				let { res_ok, res } = this.mono_wasm_set_value_on_object_info (id.o, name, newvalue);
+				if (!res_ok)
+					throw new Error (`Invoking setter on ${objectIdStr} failed`);
+
+				getter_res = res;
```

Comment:
There are no properties result here. Once you use `_register_c_fn`, you should have just the `bool` result from the C function.